concordium-consensus/src/Concordium/Kontrol/Bakers.hs

```diff
@@ -179,7 +180,7 @@ computeBakerStakesAndCapital ::
     PoolParameters' 'PoolParametersVersion1 ->
     [ActiveBakerInfo m] ->
     [ActiveDelegatorInfo] ->
-    BakerStakesAndCapital m
+    BakerStakesAndCapital (BakerInfoRef m)
```

Comment:
Change the signature to:
```
computeBakerStakesAndCapital ::
    PoolParameters' 'PoolParametersVersion1 ->
    [ActiveBakerInfo' bakerInfoRef] ->
    [ActiveDelegatorInfo] ->
    BakerStakesAndCapital bakerInfoRef
```
Then you don't need to `AllowAmbiguousTypes`.

Because `BakerInfoRef` is not injective, `m` is an ambiguous type variable: i.e. there can be multiple different `m`s for which the types reduce to the same thing.

Reply:
Changed it and removed the pragma.